### PR TITLE
feat: pick node to spin up a pool by tenant id

### PR DESCRIPTION
**Pull request by abc3**
To mitigate the risk of race conditions, this PR introduces a new feature that automatically selects a node based on the tenant id.

### CODE REVIEW

**Comment by chasers** [comment]
Looks good assuming tests will pass.

File: lib/supavisor/helpers.ex [hunk 1]
```diff
@@ -317,4 +317,17 @@ defmodule Supavisor.Helpers do
     |> :erlang.md5()
     |> Base.encode16(case: :lower)
   end
+
+  @spec rpc(Node.t(), module(), atom(), [any()], non_neg_integer()) :: {:error, any()} | any()
+  def rpc(node, module, function, args, timeout \\ 15_000) do
+    try do
+      :erpc.call(node, module, function, args, timeout)
+      |> case do
+        {:EXIT, _} = badrpc -> {:error, {:badrpc, badrpc}}
```
[the remark this answers is not in the document]
Comment (abc3) [reply lib/supavisor/helpers.ex:326]: Under the hood, `:erpc.call` invokes [spawn_request](https://www.erlang.org/doc/man/erlang#spawn_request-5) with the [monitor option](https://github.com/erlang/otp/blob/938c87f5caa6cc862dca2f1b2385eb1348ba7fd8/lib/kernel/src/erpc.erl#L146), which I think can produce `:EXIT` sometimes

[the remark this answers is not in the document]
Comment (wojtekmach) [reply lib/supavisor/helpers.ex:326]: Off the top of my head monitors tend to send DOWN _messages_. However if you have seen this behaviour in the wild, please carry on.

[the remark this answers is not in the document]
Comment (abc3) [reply lib/supavisor/helpers.ex:326]: I frequently see that :EXIT messages in Realtime, it uses pretty similar behavior for launching nodes. However, I'll check messages in Supavisor when this is rolled out to production

**Review by delgado3d — APPROVED** [review]

**Pull request merged**